synr/compiler.py

```diff
@@ -727,10 +727,10 @@ def my_function(x):
             else:
                 # make sure to preserve blank lines for correct spans
                 source = "\n".join([l[start_column:].rstrip() for l in lines])
-        mod = inspect.getmodule(program)
-        if mod is not None:
-            full_source = inspect.getsource(mod)
-        else:
+        try:
+            full_source, _ = inspect.findsource(program)
```

Comment:
Please add a comment block here explaining why we are doing this

Reply:
might be useful to isolate it into a sub-function _get_full_source

Reply:
give findsource is an internal API, it. might be helpful to first try getsource, catch the exception, then fall back to findsource, catch other exceptions, then use the original source